lighthouse-core/gather/gather-runner.js

```diff
@@ -106,7 +106,9 @@ class GatherRunner {
     log.log('status', 'Initializing…');
     const resetStorage = !options.settings.disableStorageReset;
     // Enable emulation based on settings
-    await driver.assertNoSameOriginServiceWorkerClients(options.requestedUrl);
```

Comment:
I wonder if it might make more sense long term to pass the settings into `Driver` and just have all the methods that only make sense for navigation be noops.

patching the holes here seems easy to miss one, OTOH it might be confusing to have `loadPage` do nothing based on some hidden context